components/raindrop/raindrop.app.mjs

```diff
@@ -1,11 +1,162 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "raindrop",
-  propDefinitions: {},
+  propDefinitions: {
+    raindropID: {
+      type: "string",
+      label: "Raindrop ID",
+      description: "Existing raindrop ID",
+    },
+    collectionID: {
+      type: "string",
+      label: "Collection ID",
+      description: "The collection ID",
+    },
+    expanded: {
+      type: "boolean",
+      label: "Expanded",
+      description: "Whether the collection's sub-collections are expanded",
+      optional: true,
+    },
+    title: {
+      type: "string",
+      label: "Title",
+      description: "Name of the collection",
+    },
+    sort: {
+      type: "integer",
+      label: "Sort",
+      description: "The order of collection (descending). Defines the position of the collection among all the collections with the same parent ID",
+      optional: true,
+    },
+    public: {
+      type: "boolean",
+      label: "Public",
+      description: "Collection and raindrops that it contains will be accessible without authentication?",
+      optional: true,
+    },
+    view: {
+      type: "string",
+      label: "View",
+      description: "View style of collection. Defaults to list",
+      optional: true,
+      options() {
+        return [
+          "list",
+          "simple",
+          "grid",
+          "masonry",
+        ];
+      },
+    },
+    cover: {
+      type: "string[]",
+      label: "Cover",
+      description: "Collection cover url",
+      optional: true,
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async _makeRequest($, opts) {
+      if (!opts.headers) opts.headers = {};
+      opts.headers["user-agent"] = "@PipedreamHQ/pipedream v0.1";
+      if (!opts.method) opts.method = "get";
+      const { path } = opts;
+      delete opts.path;
+      opts.url = `https://api.raindrop.io/rest/v1${path[0] === "/"
+        ? ""
+        : "/"}${path}`;
+      try {
+        return await axios($ ?? this, opts);
+      } catch (err) {
+        this._throwFormattedError(err);
+      }
+    },
```

Comment:
Can you refactor this function with
```suggestion
    async _makeRequest($ = this, opts) {
      const {
        method = "get",
        path,
        data,
        params,
        ...otherOpts
      } = opts;
      return axios($, {
        method,
        url: `https://api.raindrop.io/rest/v1${path}`,
        headers: {
          "user-agent": "@PipedreamHQ/pipedream v0.1",
          ...this._getHeaders(),
        },
        data,
        params,
        ...otherOpts,
      });
    },
```
and I think you can remove `_throwFormattedError` because Pipedream's `axios` lib will handle all errors for us.